Formula/dmd.rb

```diff
@@ -77,63 +53,31 @@ def install
-          EOS
-      etc.install conf
-      install_new_dmd_conf
-
-      make_args.unshift "DMD=#{bin}/dmd"
-
-      (buildpath/"druntime").install resource("druntime")
-      (buildpath/"phobos").install resource("phobos")
-
-      system "make", "-C", "druntime", *make_args
-      system "make", "-C", "phobos", "VERSION=#{buildpath}/VERSION", *make_args
-
-      (include/"dlang/dmd").install Dir["druntime/import/*"]
-      cp_r ["phobos/std", "phobos/etc"], include/"dlang/dmd"
-      lib.install Dir["druntime/lib/*", "phobos/**/libphobos2.a"]
-
-      resource("tools").stage do
-        inreplace "posix.mak", "install: $(TOOLS) $(CURL_TOOLS)", "install: $(TOOLS) $(ROOT)/dustmite"
-        system "make", "install", *make_args
-      end
-    else
-      make_args.unshift "DMD_DIR=#{buildpath}", "DRUNTIME_PATH=#{buildpath}/druntime", "PHOBOS_PATH=#{buildpath}/phobos"
-      (buildpath/"druntime").install resource("druntime")
-      (buildpath/"phobos").install resource("phobos")
-      system "make", "-C", "druntime", *make_args
-      system "make", "-C", "phobos", "VERSION=#{buildpath}/VERSION", *make_args
-
-      resource("tools").stage do
-        inreplace "posix.mak", "install: $(TOOLS) $(CURL_TOOLS)", "install: $(TOOLS) $(ROOT)/dustmite"
-        system "make", "install", *make_args
-      end
-
-      (include/"dlang/dmd").install Dir["druntime/import/*"]
-      cp_r ["phobos/std", "phobos/etc"], include/"dlang/dmd"
-      lib.install Dir["druntime/lib/*", "phobos/**/libphobos2.a"]
-
-      (buildpath/"dmd.conf").write <<-EOS.undent
-        [Environment]
-        DFLAGS=-I#{opt_include}/dlang/dmd -L-L#{opt_lib}
-      EOS
-      etc.install "dmd.conf"
+    make_args.unshift "DMD_DIR=#{buildpath}", "DRUNTIME_PATH=#{buildpath}/druntime", "PHOBOS_PATH=#{buildpath}/phobos"
+    (buildpath/"druntime").install resource("druntime")
+    (buildpath/"phobos").install resource("phobos")
+    system "make", "-C", "druntime", *make_args
+    system "make", "-C", "phobos", "VERSION=#{buildpath}/VERSION", *make_args
+
+    resource("tools").stage do
+      inreplace "posix.mak", "install: $(TOOLS) $(CURL_TOOLS)", "install: $(TOOLS) $(ROOT)/dustmite"
+      system "make", "install", *make_args
     end
+
+    (include/"dlang/dmd").install Dir["druntime/import/*"]
+    cp_r ["phobos/std", "phobos/etc"], include/"dlang/dmd"
+    lib.install Dir["druntime/lib/*", "phobos/**/libphobos2.a"]
+
+    (buildpath/"dmd.conf").write <<-EOS.undent
+      [Environment]
+      DFLAGS=-I#{opt_include}/dlang/dmd -L-L#{opt_lib}
+    EOS
+    etc.install "dmd.conf"
   end
 
   # Previous versions of this formula may have left in place an incorrect
   # dmd.conf.  If it differs from the newly generated one, move it out of place
   # and warn the user.
-  # This must be idempotent because it may run from both install() and
-  # post_install() if the user is running `brew install --build-from-source`.
   def install_new_dmd_conf
```

Comment:
Move the content of `install_new_dmd_conf` directly into `post_install`, since now it isn't called from any other place anymore.